backends/gdx-backend-android/src/com/badlogic/gdx/backends/android/AndroidGL20.java

```diff
@@ -288,31 +288,27 @@ public int glGenTexture () {
 		return ints[0];
 	}
 
-	public String glGetActiveAttrib(int program, int index, IntBuffer size, Buffer type) {
-		//it is assumed that size and type are both int buffers of length 1 with a single integer at position 0
-		
-		//length
-		ints[0] = 0;
+	public String glGetActiveAttrib(int program, int index, IntBuffer size, IntBuffer type) {
+		GLES20.glGetActiveAttrib(program, index, buffer.length, ints, 0, ints2, 0, ints3, 0, buffer, 0);
+
 		//size
-		ints2[0] = size.get(0);
+		size.put(ints2[0]);

```

Comment:
I think you should use absolute puts (with index 0) rather than relative. I think that is the convention? (The gets that this replaced were absolute.)

Reply:
I am not sure if it is Java convention to use `Buffer`s absolutely. Many methods in the standard library use `Buffer`s relatively: for example `ByteChannel.read` or `.write`.
Other backends also do relative puts:
 * GWT [does relative puts](https://github.com/libgdx/libgdx/blob/master/backends/gdx-backends-gwt/src/com/badlogic/gdx/backends/gwt/GwtGL20.java#L551)
 * RoboVM/Moe seem to do realtive puts: the JNI binding for glGetActiveUniform [calls getDirectBufferPointer](https://github.com/libgdx/libgdx/blob/master/gdx/jni/iosgl/iosgl20.cpp#L855), which [returns the address of the current position in the buffer](https://github.com/libgdx/libgdx/blob/master/gdx/jni/iosgl/iosgl20.cpp#L89), as far as I understand.
 * LWJGL [does relative (but incorrect) puts](https://github.com/libgdx/libgdx/blob/master/backends/gdx-backend-lwjgl/src/com/badlogic/gdx/backends/lwjgl/LwjglGL20.java#L359)
 * LWJGL3 does relative puts: `GL20C` [calls `MemoryUtils.memAddress`](https://github.com/LWJGL/lwjgl3/blob/master/modules/lwjgl/opengl/src/generated/java/org/lwjgl/opengl/GL20C.java#L1030), which [returns the address of the current position in the buffer](https://github.com/LWJGL/lwjgl3/blob/master/modules/lwjgl/core/src/main/java/org/lwjgl/system/MemoryUtil.java#L780)

I changed it to absolute puts for now, but I think maybe relative puts would be better.

Reply:
I thought it was a libGDX convention, but I don't know what is the correct way to do this. Anyway, good job.